flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/FileStoreSource.java

```diff
@@ -93,17 +93,22 @@ public Boundedness getBoundedness() {
             read.withDropDelete(false);
         }
 
+        int[][] valueCountProject = null;
         if (projectedFields != null) {
             if (valueCountMode) {
-                // TODO when isContinuous is false, don't project keys, and add key projection to
-                // split reader
-                read.withKeyProjection(projectedFields);
+                // when isContinuous is false, don't project keys, otherwise, there is no way to
```

Comment:
Nit: personally, I think the current comment is not so straightforward. What about "push projection to file store for better performance under continuous read mode, because the merge cannot be performed anyway"